lib/diagnostic.rb

```diff
@@ -7,36 +7,40 @@ class Diagnostic
   # Assume each entry in the list implements a length method that returns
   # a Fixnum.
   def longest(list)
-    # Your code goes here
+    list.reduce { |long, word|
+      if word.length >= long.length
+        long = word
+      end
+    }
```

Comment:
Got a little verbose here, there's usually always a quick way to do things in Ruby like this:

`list.max_by(&:length)`